src/go/rpk/pkg/cli/cmd/wasm/deploy.go

```diff
@@ -0,0 +1,147 @@
+package wasm
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/utils"
+)
+
+func NewDeployCommand(
+	fs afero.Fs, createProducer func(bool, int32) (sarama.SyncProducer, error),
+) *cobra.Command {
+	var description string
+
+	command := &cobra.Command{
+		Use:	"deploy <path>",
+		Short:	"deploy inline WASM function",
+		Args:	cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			path := args[0]
+			fullFileName := filepath.Base(path)
+			fileExt := filepath.Ext(fullFileName)
+			// validate file extension, just allow js extension
+			if fileExt != ".js" {
+				return fmt.Errorf("can't deploy '%s': only .js files are supported.", path)
+			}
+			fileName := strings.Replace(fullFileName, fileExt, "", 1)
+			fileContent, err := utils.ReadFileLines(fs, path)
+			if err != nil {
+				return err
+			}
+			var bytes []byte
+			for _, v := range fileContent {
+				bytes = append(bytes, []byte(v)...)
+			}
+			// create producer
+			producer, err := createProducer(false, -1)
+
+			if err != nil {
+				return err
+			}
+			return deploy(
+				fileName,
+				bytes,
+				description,
+				producer,
+			)
+		},
+	}
+
+	command.Flags().StringVar(
+		&description,
+		"description",
+		"",
+		"description about what or how wasm function do, in order to "+
+			"give some information to another developers",
+	)
+
+	return command
+}
+
+func deploy(
+	fileName string,
+	fileContent []byte,
+	description string,
+	producer sarama.SyncProducer,
+) error {
+	// create headers
+	headers, err := createHeaderDeployMessage("deploy", description, fileContent)
+	if err != nil {
+		return err
+	}
+	// create message
+	messageContent, err := utils.WriteBuffer([]byte{}, fileContent)
+	if err != nil {
+		return err
+	}
+	// publish message
+	error := kafka.PublishMessage(messageContent, fileName, producer, kafka.CoprocessorTopic, headers)
+	if error != nil {
+		return fmt.Errorf("Error on deploy %s.js, message error: %s, please check "+
+			"your redpanda instance ", fileName, error)
+	}
+	return nil
+}
+
+func createHeaderDeployMessage(
+	action string, description string, content []byte,
+) ([]sarama.RecordHeader, error) {
+	var headersResult []sarama.RecordHeader
+	// create simple struct for key string and value string
+	value := []struct {
+		key	string
+		value	string
+	}{
+		{key: "action", value: "deploy"},
```

Comment:
I agree with Rob. This looks like a mistake

Reply:
Yes, it is a mistake, I should use the action variable